src/audio/mixin_mixout/mixin_mixout.c

```diff
@@ -182,46 +182,17 @@ static int mix_and_remap(struct comp_dev *dev, const struct mixin_data *mixin_da
 
 	sink_config = &mixin_data->sink_config[sink_index];
 
-	if (sink_config->mixer_mode == IPC4_MIXER_NORMAL_MODE) {
-		/* Mix streams. mix_channel() is reused here to mix streams, not individual
-		 * channels. To do so, (multichannel) stream is treated as single channel:
-		 * channel count is passed as 1, channel index is 0, frame indices (start_frame
-		 * and mixed_frame) and frame count are multiplied by real stream channel count.
-		 */
-		mixin_data->normal_mix_channel(sink, start_frame * audio_stream_get_channels(sink),
-					       mixed_frames * audio_stream_get_channels(sink),
-					       source,
-					       frame_count * audio_stream_get_channels(sink),
-					       sink_config->gain);
-	} else if (sink_config->mixer_mode == IPC4_MIXER_CHANNEL_REMAPPING_MODE) {
-		int i;
-
-		for (i = 0; i < audio_stream_get_channels(sink); i++) {
-			uint8_t source_channel =
-				(sink_config->output_channel_map >> (i * 4)) & 0xf;
-
-			if (source_channel == 0xf) {
-				mixin_data->mute_channel(sink, i, start_frame, mixed_frames,
-							 frame_count);
-			} else {
-				if (source_channel >= audio_stream_get_channels(source)) {
-					comp_err(dev, "Out of range chmap: 0x%x, src channels: %u",
-						 sink_config->output_channel_map,
-						 audio_stream_get_channels(source));
-					return -EINVAL;
-				}
-				mixin_data->remap_mix_channel(sink, i,
-							      audio_stream_get_channels(sink),
-							      start_frame, mixed_frames,
-							      source, source_channel,
-							      audio_stream_get_channels(source),
-							      frame_count, sink_config->gain);
-			}
-		}
-	} else {
-		comp_err(dev, "Unexpected mixer mode: %d", sink_config->mixer_mode);
-		return -EINVAL;
-	}
+	/* Mix streams. mix_channel() is reused here to mix streams, not individual
+	 * channels. To do so, (multichannel) stream is treated as single channel:
+	 * channel count is passed as 1, channel index is 0, frame indices (start_frame
+	 * and mixed_frame) and frame count are multiplied by real stream channel count.
+	 */
+	mixin_data->normal_mix_channel(sink, start_frame * audio_stream_get_channels(sink),
```

Comment:
normal_mix_channel() originally was implemented to be used in both normal mode and channel remapping mode. It mixed one channel from one stream with another channel from another stream. In normal mode (without remapping, streams have same number of channels) fake channel count was used equal to 1: so streams were treated as single channel streams and were mixed entirely at one call. Hence that comment describing this trick.
However, I see, some time ago that channel mixing feature was removed and current version of normal_mix_channel() only mixes entire streams.
So that comment above can be removed (should be removed previously). That function could also be renamed, for example, to just "mix()" as now it has nothing to do with channels (it iterates by samples and does not care about channels) and now "normal" is the only supported mode.